serverless/serverless
# Added autocomplete to the CLI with support for sub commands & options

## What did you implement:

Closes #3254 


Added autocomplete support to the CLI.

## How did you implement it:
Using `tabtab` package, on Serverless installation, a script runs that auto-detects your shell type (ie. bash, fish..etc) and installs the completion function into that shell. The shell completion function logic is written in the new `autocomplete.js` file.


## How can we verify it:

1. Install this branch globally by fetching it and running `npm install -g https://github.com/serverless/serverless#autocomplete`. This simulates the user installing the published version.
2. restart your shell, or source is with `source ~/.bashrc` if you're on bash.
3. Run any serverless command to generate the first cache file => `serverless help`
4. Run `serverless de<tab>`, `serverless deploy fu<tab>`, `serverless inv<tab>`, `serverless invoke --st<tab>` and a lot more! you should see auto complete working with command and option completion.


## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
